lib/screens/login/login_view_model.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final loginViewModelProvider =
+    StateNotifierProvider.autoDispose<LoginViewModel, void>((_) {
+  return LoginViewModel();
+});
+
+class LoginViewModel extends StateNotifier<void> {
+  LoginViewModel() : super([]);
+
+  bool validateEmail(String? email) {
```

Comment:
It would be better if we name it `isEmailValid`, since it returns the bool result 🤔